shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1566,32 +1556,51 @@ - (BOOL)isPresentingViewController {
   pointer_data.Clear();
 
   pointer_data.kind = flutter::PointerData::DeviceKind::kMouse;
-  pointer_data.change = _mouseState.flutter_state_is_added ? flutter::PointerData::Change::kAdd
-                                                           : flutter::PointerData::Change::kHover;
-  pointer_data.pointer_identifier = reinterpret_cast<int64_t>(_pointerInteraction.get());
+  switch (_hoverGestureRecognizer.get().state) {
+    case UIGestureRecognizerStateBegan:
+      pointer_data.change = flutter::PointerData::Change::kAdd;
+      break;
+    case UIGestureRecognizerStateChanged:
+      pointer_data.change = flutter::PointerData::Change::kHover;
+      break;
+    case UIGestureRecognizerStateEnded:
+    case UIGestureRecognizerStateCancelled:
+      pointer_data.change = flutter::PointerData::Change::kRemove;
+      break;
+    default:
+      FML_LOG(ERROR) << "Unhandled hover phase: " << _hoverGestureRecognizer.get().state;
```

Comment:
This log is not user actionable. Let's remove it. Also, `flutter::PointerData::Change::kCancel` seems more apt here.

Reply:
kCancel is meant for when the pointer is down, since this pointer is never down it will cause failed assertions during processing. 
Since the remaining unhandled and unexpected states are all a kind of semi-recognized I thought that kHover was the closest choice. And it's the only one that will not cause a fatal double add or double remove.